### PR TITLE
Update seek branch with latest changes from upstream v1.1.0

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff https://github.com/hashicorp/go-immutable-radix/compare/8aac2701530899b64bdea735a1de8da899815220...v1.1.0

relevant changes:

- https://github.com/hashicorp/go-immutable-radix/pull/15 Fix Len() when deleting keys in a transaction
- https://github.com/hashicorp/go-immutable-radix/pull/24 Add LowerBound to allow range scans

### CODE REVIEW

**Comment by thaJeztah** [comment]
@tonistiigi

**Comment by tonistiigi** [comment]
@thaJeztah I think I'll rebase instead so we can keep the history clean? sgty?

**Comment by thaJeztah** [comment]
We should tag the current version otherwise the commit that's being vendored will disappear; that's why I went for a merge instead of a rebase

**Comment by thaJeztah** [comment]
@tonistiigi @samwhited PTAL; updated this to the upstream v1.1.0 tag. Looks like without those changes, other dependencies (go-memdb) cannot be updated https://github.com/docker/swarmkit/pull/2905#issuecomment-534996944

**Comment by SamWhited** [comment]
LGTM, but out of curiosity how was this PR created? If I checkout Tonis' current master and merge the tag I get a diff against this branch (it appears to be missing Tonis' commits adding `Seek()`).

**Comment by thaJeztah** [comment]
hm. interesting; I checked out the `seek` branch from tonis's fork and did a `git merge v1.1.0`. Let me try again to be sure I didn't mess up 🤔

**Comment by SamWhited** [comment]
I'm probably on the wrong branch, I just saw that this was based on master and used that and didn't see the seek branch.

**Comment by thaJeztah** [comment]
oh! ah, yes, that would explain. We should probably mark the `seek` branch as protected to be sure it doesn't get deleted and such, or re-consider upstreaming this change (which of course would be easier to manage it)

**Comment by SamWhited** [comment]
This PR is merging into `master`, not `seek`. Was that intentional?

**Comment by thaJeztah** [comment]
oh, wow; good catch. how did that change? 🤔 let me fix that

**Comment by SamWhited** [comment]
Now everything is fine without a diff; LGTM!

**Comment by tonistiigi** [comment]
Still thinking this should be a rebase. We can keep the old branch.

**Comment by thaJeztah** [comment]
should we tag the old version, then we don't need to keep the branch around

**Comment by AkihiroSuda** [comment]
Can we rather propose the changes to the upstream?

**Comment by thaJeztah** [comment]
> Can we rather propose the changes to the upstream?

I'm definitely +1 for that; @tonistiigi I know I asked in the past, but forgot; is there any reason for not opening https://github.com/tonistiigi/go-immutable-radix/commit/826af9ccf0feeee615d546d69b11f8e98da8c8f1 as a pull request upstream?

**Comment by thaJeztah** [comment]
ping @tonistiigi

**Comment by thaJeztah** [comment]
Should no longer be needed, now that https://github.com/moby/buildkit/pull/2182 was merged
